lib/applyToDefaults.js

```diff
@@ -73,7 +73,7 @@ internals.applyToDefaultsWithShallow = function (defaults, source, options) {
 internals.reachCopy = function (dst, src, path) {
 
     for (const segment of path) {
-        if (!(segment in src)) {
+        if (src[segment] === undefined) {
```

Comment:
This does not fix the deeper issue here. Any falsy value will cause a throw.

Reply:
I just checked `0`, `false`, `NaN`, `''`, and `null` - and only `null` made it throw. I've updated the code to also work for `null`.

Reply:
Hmm not sure what I was doing when I tested this. But any non-object will fail on this - not even just falsy values. Also a regular string value will make it fail. I've updated the code so now only objects are allowed to continue the loop.